### DISCUSSION TITLE
widget player security issue.

### DISCUSSION THREAD

**Discussion by fsan2val** [AzuraCast/AzuraCast#6547] [Ideas]
If you embed the player in any website, you can't see the cover in the form of a popup like the default public page, but rather it takes you to the image directory link, which causes security problems due to the azuracast installation path

IMG: https://ibb.co/FqC0wK3

**Comment by Moonbase59** [comment]
I wouldn’t see that as a "security issue"—some links like the station, the stream address and the covers _must_ be available. The cover image location you could always see.

Someone knowing AzuraCast’s structure would know where to look anyway. So as long as all links and the API are protected enough, it should be okay.

![Nite Radio – Mozilla Firefox_014](https://github.com/AzuraCast/AzuraCast/assets/3706922/61ca0644-7f36-4a07-8725-abb7911bab41)

**Reply by fsan2val** [reply]
Perfect, at a good time, it's not a security problem. What does catch my attention is that the pop up of the cover image that does not work using the player as a widget on another website rather takes you to the path of the image

![azuracast popup](https://github.com/AzuraCast/AzuraCast/assets/69484466/2215642e-aa68-47c5-9b42-916af9538eb0)

**Comment by BusterNeece** [comment]
Any determined user could just look at the URL of the embedded iframe at any point. Just knowing the URL of the AzuraCast installation doesn't convey any security problem, nor can users just look through media hosted on your installation, because as you can see, the media IDs are all unique identifiers and not sequential.

The fact that the album art doesn't enlarge to be bigger in the embedded pages is also by design, simply because there isn't enough room to show the enlarged version inside an iframe, and there's no way for us to control the outer page when embedded like that.

**Reply by fsan2val** [reply]
Thanks for your guidance, but it seems strange to me that the pop up in the widgets doesn't work, I think this is an azuracast error. greetings and thanks again

**Reply by BusterNeece** [reply]
It is by design.
